lexers/mediawiki.lua

```diff
@@ -1,42 +1,72 @@
 -- Copyright 2006-2025 Mitchell. See LICENSE.
 -- MediaWiki LPeg lexer.
 -- Contributed by Alexander Misel.
-
-local lexer = require('lexer')
-local token, word_match = lexer.token, lexer.word_match
+local lexer = lexer
 local P, S, B = lpeg.P, lpeg.S, lpeg.B
+local lex = lexer.new(...)
 
-local lex = lexer.new('mediawiki')
+-- HTML-like tags
+local dq_str = P('"') * ((lexer.any - S('>"\\')) + ('\\' * lexer.any))^0 * P('"')
+local unquoted_attr = (lexer.any - (S('"' .. "'" .. '<>=') + lexer.space))^1
+local tag_attr = lex:tag(lexer.ATTRIBUTE, lexer.alpha^1 * lexer.space^0 *
+	('=' * lexer.space^0 * (dq_str + unquoted_attr))^-1 * lexer.space^0)
+local tag_name = lexer.alnum^1
+local tag_inner_content = lexer.space^0 * tag_attr^0 * lexer.space^0 * P('/')^-1
 
--- Comments.
-lex:add_rule('comment', token(lexer.COMMENT, lexer.range('<!--', '-->')))
+-- The tag rule should consume from '<' to the final '>'
+lex:add_rule('tag', lex:tag(lexer.TAG, '<' * P('/')^-1 * tag_name * tag_inner_content * P('>')))
```

Comment:
Because you are tagging the contents of `tag_inner_content` as a `TAG`, `tag_attr` will never be able to tag attributes as `ATTRIBUTE` even though it tries to.

I thought the previous `tag_start * tag_attr^0 * tag_end` did the right thing. Was that not the case?